src/transformStateWithClones.js

```diff
@@ -7,7 +7,32 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let myState = { ...state };
+  const result = [];
+
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      myState = {
+        ...myState,
+        ...obj.extraData,
+      };
+      result.push({ ...myState });
+    }
+
+    if (obj.type === 'removeProperties') {
+      for (const toDelete of obj.keysToRemove) {
+        delete myState[toDelete];
+      }
+      result.push({ ...myState });
+    }
+
+    if (obj.type === 'clear') {
+      myState = {}; // Очищаємо myState
+      result.push({ ...myState });
+    }
```

Comment:
Good work but let's rewrite it using switch/case as we rely on single `obj.type` value